mitmproxy/tools/web/app.py

```diff
@@ -290,13 +292,44 @@ def get(self):
 class WebSocketEventBroadcaster(tornado.websocket.WebSocketHandler):
     # raise an error if inherited class doesn't specify its own instance.
     connections: ClassVar[set[WebSocketEventBroadcaster]]
+    application: Application  # reference to the application, in order to be able to update the filters
 
     def open(self, *args, **kwargs):
         self.connections.add(self)
 
     def on_close(self):
         self.connections.discard(self)
 
+    async def on_message(self, message: str):
+        try:
+            data = json.loads(message)
+
+            # Validate the required fields
+            if not all(key in data for key in ["resource", "cmd", "name", "expr"]):
+                print(f"Invalid message format: Missing required fields.")
+                return
+
+            resource = data["resource"]
+            command = data["cmd"]
+            name = data["name"]
+            expression = data["expr"]
+
+            if resource == "flows" and command == "updateFilter":
+                filters_manager = self.application.master.filters_manager
```

Comment:
Sharing one filter manager across all client connections is a bad idea. For example, two different clients may use two different "search" filters at the same time. These would collide right now.


I think it makes most sense to track filter state as part of the `ClientConnection` instance. ClientConnection needs dedicated methods like `broadcast_flow_update(f: Flow)` so that it can just apply the filter.

Reply:
Good point! I also removed the broadcasting function when sending the matching flow IDs; otherwise, every client connection would have received updates meant for other clients, making it impossible to maintain different search filters for each client